youtube_dl/extractor/tvp.py

```diff
@@ -250,3 +250,114 @@ def _real_extract(self, url):
         display_id, playlist_id = mobj.group('display_id', 'id')
         return self.playlist_result(
             self._entries(display_id, playlist_id), playlist_id)
+
+
+class TVPappIE(InfoExtractor):
+    IE_NAME = 'tvp:app'
+    # https://vod.tvp.pl/seriale,18/korona-krolow-jagiellonowie-odcinki,292227/odcinek-37,S01E37,392532
+    # https://vod.tvp.pl/programy,88/korona-krolow-taka-historia-odcinki,283653/odcinek-1,S02E01,386064
+    # https://vod.tvp.pl/filmy-dokumentalne,163/jurek,376503
+    # https://vod.tvp.pl/dla-dzieci,24/zubr-pompik-odcinki,282014/odcinek-1,S01E01,319853
+    # https://vod.tvp.pl/teatr-telewizji,202/barwy-uczuc,392420
+    # https://vod.tvp.pl/filmy-fabularne,136/rozlam,390638
+    _VALID_URL = r'https://vod.tvp.pl/(seriale|filmy-fabularne|programy|filmy-dokumentalne|dla-dzieci|teatr-telewizji|informacje-i-publicystyka),[0-9]+/([^,]+,[0-9]+/)?[^,]+,([^,]+,)?(?P<id>[0-9]+)'
```

Comment:
This pattern needs to be disjoint from the `TVPIE._VALID_URL`.

If not, `TVPIE` needs this:
```py
    @classmethod
    def suitable(cls, url):
        return not TVPAppIE.suitable(url) and super(TVPIE, cls).suitable(url)
```